cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaClusterResourcesTest.java

```diff
@@ -0,0 +1,137 @@
+import io.strimzi.api.kafka.model.listener.arraylistener.KafkaListenerType;
+import io.strimzi.api.kafka.model.storage.PersistentClaimStorageBuilder;
+import io.strimzi.operator.cluster.KafkaVersionTestUtils;
+import io.strimzi.operator.common.Reconciliation;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@SuppressWarnings({"checkstyle:ClassDataAbstractionCoupling", "checkstyle:ClassFanOutComplexity", "checkstyle:JavaNCSS"})
+public class KafkaClusterResourcesTest {
+    private static final KafkaVersion.Lookup VERSIONS = KafkaVersionTestUtils.getKafkaVersionLookup();
+    private static final String NAMESPACE = "test";
+    private static final String CLUSTER = "foo";
+    private static final int REPLICAS = 3;
+    private static final Kafka KAFKA = new KafkaBuilder()
+            .withNewMetadata()
+                .withName(CLUSTER)
+                .withNamespace(NAMESPACE)
+            .endMetadata()
+            .withNewSpec()
+                .withNewZookeeper()
+                    .withReplicas(REPLICAS)
+                    .withNewPersistentClaimStorage()
+                        .withSize("100Gi")
+                    .endPersistentClaimStorage()
+                .endZookeeper()
+                .withNewKafka()
+                    .withReplicas(REPLICAS)
+                    .withListeners(new GenericKafkaListenerBuilder()
+                                    .withName("plain")
+                                    .withPort(9092)
+                                    .withType(KafkaListenerType.INTERNAL)
+                                    .withTls(false)
+                                    .build(),
+                            new GenericKafkaListenerBuilder()
+                                    .withName("tls")
+                                    .withPort(9093)
+                                    .withType(KafkaListenerType.INTERNAL)
+                                    .withTls(true)
+                                    .build())
+                    .withNewJbodStorage()
+                        .withVolumes(new PersistentClaimStorageBuilder().withId(0).withSize("100Gi").withDeleteClaim(false).build())
+                    .endJbodStorage()
+                    .withConfig(Map.of("log.message.format.version", "3.0", "inter.broker.protocol.version", "3.0"))
+                .endKafka()
+            .endSpec()
+            .build();
+
+    private record ResourcesCombo(String cpuRequest, String cpuLimit, String memoryRequest, String memoryLimit, String error) { }
+    
+    private static List<ResourcesCombo> resourcesCombos() {
+        return Arrays.asList(
```

Comment:
One particular area I'm missing here is testing different units against each other. E.g.:
* Memory request 2048Mi versus Memory limit 4Gi and the other way around
* CPU request 500m versus CPU limit 2 and the other way around